engine/src/settings/settings.cpp

```diff
@@ -96,6 +108,120 @@ void Settings::merge(const Settings& settingsToMerge)
     }
 }
 
+static inline void saveSettingsAsJson(const Settings& settings, Stream& stream, int indent)
+{
+    auto json = nlohmann::json();
+    for (const auto& value : settings.getValues())
+    {
+        char* pEnd;
+        double res = strtod(value.second.c_str(), &pEnd);
+        // Store as string if second value is not number nor boolean
+        if (pEnd == value.second.c_str() + value.second.length() && !value.second.empty())
+        {
+            json[value.first] = res;
+        }
+        else if (value.second == "true" || value.second == "false")
+        {
+            json[value.first] = value.second == "true";
+        }
+        else
+        {
+            json[value.first] = value.second;
+        }
+    }
+    stream.print(json.dump(indent));
+}
+
+bool Settings::save(std::string_view path, int indent) const
+{
+    // If the settings file is not mounted.
+    if (FileSystem::find(path) == nullptr)
+    {
+        return false;
+    }
+
+    auto stream = FileSystem::open(path, File::OpenMode::Write);
```

Comment:
I think you shoud just check if the `stream` returned by `open` is `nullptr`.
E.g., the path may point to a directory instead of a file. In that case, this will succeed, but open will return nullptr